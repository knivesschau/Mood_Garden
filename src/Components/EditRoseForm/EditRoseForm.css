/* mobile first style, min 320px */

.Rose_Editor, .Thorn_Editor, .Bud_Editor {
    box-shadow: 0px 1px 5px 2px #6E7C51; 
    border-radius: 2em;
    margin: 1em;
    padding: .50em;
    background-color: rgb(203, 211, 167, .90);
}

#edit-header {
    margin-top: .50em;
    font-size: 1.75em;
}

#rose-edit-header, #thorn-edit-header, #bud-edit-header {
    color: #6B1A1A;
    font-weight: bold;
    font-size: 1.5em;
}

#update-garden, #cancel-update {
    display: inline-block;
    box-sizing: border-box;
    font-size: 1em;
    font-weight: bold;
    border: 2px solid rgba(255,255,255,0);
    border-radius: 2em;
    padding: .5em;
    margin-right: 1em;
    margin-bottom: .75em;
    background-color: #C34444;
    color: #ffffff;
    text-shadow: 0 .65px 0.65px rgb(0,0,0,0.35);
}

#update-garden:hover, #cancel-update:hover {
    cursor: pointer;
    border-color: rgb(255,255,255, 1);
}

#update-garden:hover, #cancel-update:hover {
    background-color: #8F0000;
}

@media screen and (min-device-width: 414px) {
    .Rose_Editor, .Thorn_Editor, .Bud_Editor {
        margin: 1.5em;
    }

    #edit-header {
        font-size: 2em;
    }
    
    #rose-edit-header, #thorn-edit-header, #bud-edit-header {
        font-size: 1.75em;
    }
    
    #update-garden, #cancel-update {
        font-size: 1.10em;
    }
}

@media only screen and (min-device-width: 768px) {
    .Rose_Editor, .Thorn_Editor, .Bud_Editor {
        margin-left: 7.5em;
        margin-right: 7.5em;
    }

    #edit-header {
        margin-top: 1em;
    }

    #update-garden, #cancel-update {
        margin-left: -.50em;
        margin-right: 1.5em;
    }
}

@media only screen and (min-device-width: 1000px) {
    .Rose_Editor, .Thorn_Editor, .Bud_Editor {
        margin-left: 15em;
        margin-right: 15em;
    }

    #edit-header {
        font-size: 2.75em;
    }

    #rose-edit-header, #thorn-edit-header, #bud-edit-header {
        font-size: 2.5em;
    }

    #update-garden, #cancel-update {
        font-size: 1.5em;
    }
}

@media only screen and (min-device-width: 1200px) {
    .Rose_Editor, .Thorn_Editor, .Bud_Editor {
        margin-left: 20em;
        margin-right: 20em;
    }
}

@media only screen and (min-device-width: 1700px) {
    .Rose_Editor, .Thorn_Editor, .Bud_Editor {
        margin-left: 35em;
        margin-right: 35em;
    }
}