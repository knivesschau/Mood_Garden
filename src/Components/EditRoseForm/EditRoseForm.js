import React, {Component} from 'react';

export default class EditRoseForm extends Component {
    render() {
        const {rose, thorn, bud, color, entry_date} = this.props;

        return (
            <>
            <h1>Editing {color} Rose Planted on {new Date(entry_date).toLocaleDateString()}</h1>

            <div className="Rose_Editor">
                <h2>Rose</h2>

                <p id="rose-prompt">What is something you are looking forward to? Are there any new ideas that you would like to see grow?</p>
                
                <textarea name="rose-text" id="rose-entry-input" rows="15" placeholder={rose}/>
                
                </div>

            <div className="Thorn_Editor">
                <h2>Thorn</h2>

                <p id="thorn-prompt">What didn't go as planned? What stressed you out or made you feel upset?</p>

                <textarea name="thorn-text" id="thorn-entry-input" rows="15" placeholder={thorn}/>
            </div>

            <div className="Bud_Editor">
                <h2>Bud</h2>

                <p id="bud-prompt">What is something you are looking forward to? Are there any new ideas that you would like to see grow?</p>
            
                <textarea name="bud-text" id="bud-entry-input" rows="15" placeholder={bud}/>
            </div>
            </>
        );
    }
}