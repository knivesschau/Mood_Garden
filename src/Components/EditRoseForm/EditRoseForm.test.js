import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import EditRoseForm from './EditRoseForm';

describe ('EditRoseForm Component', () => {
    it ('renders the complete EditRoseForm Component', () => {
        const wrapper = shallow(<EditRoseForm/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    })
})