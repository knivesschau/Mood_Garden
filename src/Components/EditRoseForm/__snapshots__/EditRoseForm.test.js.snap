// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRoseForm Component renders the complete EditRoseForm Component 1`] = `
<Fragment>
  <h1
    id="edit-header"
  >
    Editing 
     Rose Planted on 
    Invalid Date
  </h1>
  <div
    className="Rose_Editor"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      color="#FFFAEA"
      fixedWidth={false}
      flip={null}
      icon="grin-beam"
      id="rose-icon"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="3x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <h2
      id="rose-edit-header"
    >
      Rose
    </h2>
    <label
      htmlFor="rose-entry-input"
    >
      <p
        id="rose-prompt"
      >
        What brought you happiness today? Did you make progress on any ideas or goals?
      </p>
      <textarea
        id="rose-entry-input"
        name="rose-edit"
        onChange={[Function]}
        rows="15"
      />
      <ErrorValidation />
    </label>
  </div>
  <div
    className="Thorn_Editor"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      color="#A272AA"
      fixedWidth={false}
      flip={null}
      icon="frown"
      id="thorn-icon"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="3x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <h2
      id="thorn-edit-header"
    >
      Thorn
    </h2>
    <label
      htmlFor="thorn-entry-input"
    >
      <p
        id="thorn-prompt"
      >
        Was there anything that upset or frustrated you today? What were the sources of those feelings?
      </p>
      <textarea
        id="thorn-entry-input"
        name="thorn-edit"
        onChange={[Function]}
        rows="15"
      />
      <ErrorValidation />
    </label>
  </div>
  <div
    className="Bud_Editor"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      color="#8B5151"
      fixedWidth={false}
      flip={null}
      icon="grin-alt"
      id="bud-icon"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="3x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <h2
      id="bud-edit-header"
    >
      Bud
    </h2>
    <label
      htmlFor="bud-entry-input"
    >
      <p
        id="bud-prompt"
      >
        What is something you are looking forward to? What would you like to spend more time focusing on?
      </p>
      <textarea
        id="bud-entry-input"
        name="bud-edit"
        onChange={[Function]}
        rows="15"
      />
      <ErrorValidation />
    </label>
    <button
      id="update-garden"
      type="submit"
    >
      Update Rose
    </button>
    <button
      id="cancel-update"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </div>
</Fragment>
`;

exports[`EditRoseForm Component renders the correct EditRoseForm Component given props 1`] = `
<Fragment>
  <h1
    id="edit-header"
  >
    Editing 
    Yellow
     Rose Planted on 
    7/28/2020
  </h1>
  <div
    className="Rose_Editor"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      color="#FFFAEA"
      fixedWidth={false}
      flip={null}
      icon="grin-beam"
      id="rose-icon"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="3x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <h2
      id="rose-edit-header"
    >
      Rose
    </h2>
    <label
      htmlFor="rose-entry-input"
    >
      <p
        id="rose-prompt"
      >
        What brought you happiness today? Did you make progress on any ideas or goals?
      </p>
      <textarea
        defaultValue="test rose"
        id="rose-entry-input"
        name="rose-edit"
        onChange={[Function]}
        rows="15"
      />
      <ErrorValidation />
    </label>
  </div>
  <div
    className="Thorn_Editor"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      color="#A272AA"
      fixedWidth={false}
      flip={null}
      icon="frown"
      id="thorn-icon"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="3x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <h2
      id="thorn-edit-header"
    >
      Thorn
    </h2>
    <label
      htmlFor="thorn-entry-input"
    >
      <p
        id="thorn-prompt"
      >
        Was there anything that upset or frustrated you today? What were the sources of those feelings?
      </p>
      <textarea
        defaultValue="test thorn"
        id="thorn-entry-input"
        name="thorn-edit"
        onChange={[Function]}
        rows="15"
      />
      <ErrorValidation />
    </label>
  </div>
  <div
    className="Bud_Editor"
  >
    <FontAwesomeIcon
      border={false}
      className=""
      color="#8B5151"
      fixedWidth={false}
      flip={null}
      icon="grin-alt"
      id="bud-icon"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size="3x"
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
    <h2
      id="bud-edit-header"
    >
      Bud
    </h2>
    <label
      htmlFor="bud-entry-input"
    >
      <p
        id="bud-prompt"
      >
        What is something you are looking forward to? What would you like to spend more time focusing on?
      </p>
      <textarea
        defaultValue="test component"
        id="bud-entry-input"
        name="bud-edit"
        onChange={[Function]}
        rows="15"
      />
      <ErrorValidation />
    </label>
    <button
      id="update-garden"
      type="submit"
    >
      Update Rose
    </button>
    <button
      id="cancel-update"
      onClick={[Function]}
      type="button"
    >
      Cancel
    </button>
  </div>
</Fragment>
`;
